### PR TITLE
Improve tuple-related docs

### PR DESCRIPTION
**Pull request by ntrel**
* Use `auto` function return instead of ReverseTupleType as that is private.
* Document `tuple` function inside template.

### CODE REVIEW

File: std/typecons.d [hunk 1]
```diff
@@ -541,6 +543,7 @@ template Tuple(Specs...)
          *              of this `Tuple`, or can implicitly convert
          *              to those types. They must be in the same
          *              order as they appear in `Types`.
+         * See_Also: $(LREF tuple).
```
Comment (JackStouffer) [inline std/typecons.d]: This would make more sense in the top level doc for `Tuple`

File: std/typecons.d [hunk 2]
```diff
@@ -1166,15 +1169,15 @@ template Tuple(Specs...)
...
 
     Params:
         t = The `Tuple` to copy.
 
     Returns:
-        A copy of `t` with its fields in reverse order.
+        A new `Tuple`.
  */
-ReverseTupleType!T reverse(T)(T t)
+auto reverse(T)(T t)
```
Comment (JackStouffer) [inline std/typecons.d]: 👍

File: std/typecons.d [hunk 3]
```diff
@@ -1667,24 +1670,26 @@ unittest
 }
 
 /**
-    Constructs a $(D Tuple) object instantiated and initialized according to
+    Constructs a $(LREF Tuple) object instantiated and initialized according to
     the given arguments.
 
     Params:
-        Names = A list of strings naming each successive field of the `Tuple`.
+        Names = Optional list of strings naming each successive field of the `Tuple`.
```
Comment (JackStouffer) [inline std/typecons.d]: "An optional" sounds more natural to me

File: std/typecons.d (hunk 1 above)
Comment (ntrel) [inline std/typecons.d]: That comment has LREF, so you want me to add See_Also to it?

Comment (JackStouffer) [inline std/typecons.d]: yes

**Comment by ntrel** [comment]
@JackStouffer Updated with your suggestions, sorry for the delay.

**Comment by ntrel** [comment]
Ping

**Comment by JackStouffer** [comment]
Auto-merge toggled on

**Pull request merged**